### ISSUE TITLE
Issues on loading ground-truth bounding boxed

### ISSUE THREAD

**Issue by nNbSzxx** [eriklindernoren/PyTorch-YOLOv3#573]
In line 106 to line 125

    if os.path.exists(label_path):
        boxes = torch.from_numpy(np.loadtxt(label_path).reshape(-1, 5))
        # Extract coordinates for unpadded + unscaled image
        x1 = w_factor * (boxes[:, 1] - boxes[:, 3] / 2)
        y1 = h_factor * (boxes[:, 2] - boxes[:, 4] / 2)
        x2 = w_factor * (boxes[:, 1] + boxes[:, 3] / 2)
        y2 = h_factor * (boxes[:, 2] + boxes[:, 4] / 2)
        # Adjust for added padding
        x1 += pad[0]
        y1 += pad[2]
        x2 += pad[1]
        y2 += pad[3]
        # Returns (x, y, w, h)
        boxes[:, 1] = ((x1 + x2) / 2) / padded_w
        boxes[:, 2] = ((y1 + y2) / 2) / padded_h
        boxes[:, 3] *= w_factor / padded_w
        boxes[:, 4] *= h_factor / padded_h
        targets = torch.zeros((len(boxes), 6))
        targets[:, 1:] = boxes

First, in coco's annotations, the format of bounding boxes is [top-left x, top-left y, width, height]. It seems that top-left x and top-left y are treated as the central coordinates of bounding boxes in the codes.
Second, the pad adding should be pad[0], pad[2], pad[0], pad[2], not pad[0], pad[2], pad[1], pad[3].

**Comment by hiahiml** [comment]
> In line 106 to line 125
> 
> ```
> if os.path.exists(label_path):
>     boxes = torch.from_numpy(np.loadtxt(label_path).reshape(-1, 5))
>     # Extract coordinates for unpadded + unscaled image
>     x1 = w_factor * (boxes[:, 1] - boxes[:, 3] / 2)
>     y1 = h_factor * (boxes[:, 2] - boxes[:, 4] / 2)
>     x2 = w_factor * (boxes[:, 1] + boxes[:, 3] / 2)
>     y2 = h_factor * (boxes[:, 2] + boxes[:, 4] / 2)
>     # Adjust for added padding
>     x1 += pad[0]
>     y1 += pad[2]
>     x2 += pad[1]
>     y2 += pad[3]
>     # Returns (x, y, w, h)
>     boxes[:, 1] = ((x1 + x2) / 2) / padded_w
>     boxes[:, 2] = ((y1 + y2) / 2) / padded_h
>     boxes[:, 3] *= w_factor / padded_w
>     boxes[:, 4] *= h_factor / padded_h
>     targets = torch.zeros((len(boxes), 6))
>     targets[:, 1:] = boxes
> ```
> 
> First, in coco's annotations, the format of bounding boxes is [top-left x, top-left y, width, height]. It seems that top-left x and top-left y are treated as the central coordinates of bounding boxes in the codes.
> Second, the pad adding should be pad[0], pad[2], pad[0], pad[2], not pad[0], pad[2], pad[1], pad[3].

thank you. I refer to your opinion

**Comment by Flova** [comment]
This repo handles it the same as the Darknet implementation. See https://github.com/AlexeyAB/Yolo_mark/issues/60

**Comment by Flova** [comment]
This code is now rewritten. Feel free to reopen for further discussion.
